### PR TITLE
Changing build to use dotnet

### LINKED ISSUE #41: Change to build to use dotnet
At a minimum, you have to upgrade a few scripts and project.json files just like here: aspnet/Configuration@9cfd22f

cc @pranavkm @NTaylorMullen

### PR DESCRIPTION
**Pull request by pranavkm**
Fixes #41

### CODE REVIEW

**Comment by NTaylorMullen** [comment]
:shipit:

File: .travis.yml [hunk 1]
```diff
@@ -10,14 +10,11 @@ addons:
     - libssl-dev
     - libunwind8
     - zlib1g
-before_install:
-  - if test "$TRAVIS_OS_NAME" == "osx"; then brew update; brew install icu4c; fi
-env:
-  - KOREBUILD_DNU_RESTORE_CORECLR=true MONO_THREADS_PER_CPU=2000 KOREBUILD_TEST_DNXCORE=true
```
Comment (dougbu) [inline .travis.yml]: This repo doesn't hit Mono threading issues?

File: .travis.yml [hunk 2]
```diff
@@ -10,14 +10,11 @@ addons:
     - libssl-dev
     - libunwind8
     - zlib1g
-before_install:
-  - if test "$TRAVIS_OS_NAME" == "osx"; then brew update; brew install icu4c; fi
```
Comment (dougbu) [inline .travis.yml]: Better to leave these two lines alone since they'll just need to be restored later.

File: .travis.yml (hunk 1 above)
Comment (dougbu) [inline .travis.yml]: I.e. safer to preserve `MONO_THREADS_PER_CPU=2000`.

**Pull request merged**